jina/drivers/craft.py

```diff
@@ -75,7 +75,13 @@ def __call__(self, *args, **kwargs):
             ret = self.exec_fn(**_args_dict)
             if ret:
                 for k, v in ret.items():
-                    setattr(d, k, v)
+                    if k == 'blob':
+                        if isinstance(v, jina_pb2.NdArray):
+                            d.blob.CopyFrom(v)
+                        else:
+                            d.blob.CopyFrom(array2pb(v))
+                    else:
+                        setattr(d, k, v)
```

Comment:
I do see this part is related to https://github.com/jina-ai/jina/blob/ac6175a7d4ca90d8a49ae93a0f1ec61390ec0abf/jina/drivers/craft.py#L19 and some refactoring in the future could make it more consistent. After all, doc and chunk share the same `content` proto spec.

Reply:
but let's leave it as a future work